x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/action/TransportUpdateTransformAction.java

```diff
@@ -208,7 +208,17 @@ protected Response newResponse(
         List<TaskOperationFailure> taskOperationFailures,
         List<FailedNodeException> failedNodeExceptions
     ) {
-        // there should be only 1 response, todo: check
+        assert tasks.size() + taskOperationFailures.size() == 1 : "found more than one error or response";
+        if (tasks.isEmpty()) {
+            if (taskOperationFailures.isEmpty() == false) {
+                throw org.elasticsearch.ExceptionsHelper.convertToElastic(taskOperationFailures.get(0).getCause());
+            } else if (failedNodeExceptions.isEmpty() == false) {
+                throw org.elasticsearch.ExceptionsHelper.convertToElastic(failedNodeExceptions.get(0));
+            } else {
+                throw new IllegalStateException("No errors or response");
```

Comment:
good input, thanks!

There is no need to retry: This code is only hit if a transform is running and the update changes settings that are important at runtime, e.g. the setting for throttling. You are right that the task can disappear, but this case is actually benign, because than we don't need the update of runtime settings anyway. I think I have an idea how to handle this case correctly.

Reply:
@droberts195 

I've changed it now to handle 2 scenarios:

 - transform task has disappeared. This is benign and can be ignored.
 - some other error happened: I don't think this should fail the update call at this stage, because update itself was successful. Only the notification failed. If this happens the new setting has to wait until the next checkpoint, when the transform automatically reloads (you don't have to stop/start).

Because I want to backport this change to 7.17, I decided to only log the failure. As a follow up I want to add the failure to the API output only for main. Changing it in all versions would require complex BWC logic.